common/clap_utils/src/flags.rs

```diff
@@ -1,3 +1,179 @@
 //! CLI flags used across the Lighthouse code base can be located here.
+//!
+//! These flags are grouped according to which subcommands they are used in, in order to allow us to
+//! validate which flags are available to which subcommands when loading config from a file.
 
-pub const DISABLE_MALLOC_TUNING_FLAG: &str = "disable-malloc-tuning";
+use lazy_static::lazy_static;
+
+/// This macro constructs a constant for each given `(identifier, value)` tuple, and adds each constant to
+/// a `Vec` generated by `lazy_static`.  The first argument will be used as the name of the `Vec`.
+macro_rules! generate_flag_constants {
+    ($group_name:ident, $(($key:ident, $value:expr)),+) => {
+        $(pub const $key: &str = $value;)+
+        lazy_static! {
+            pub static ref $group_name: Vec<&'static str> = {
+                let mut v = vec![];
+                $(v.push($value);)+
+                v
+            };
+        }
+    }
+}
```

Comment:
You don't need to have them in a Vec, an array suffices
```suggestion
/// This macro constructs a constant for each given `(identifier, value)` tuple, and a constant
/// group containing all of those.
macro_rules! generate_flag_constants {
    ($group_name:ident, $(($key:ident, $value:expr)),+) => {
        $(pub const $key: &str = $value;)+
        pub const $group_name: &[&'static str] = &[
            $($value,)*
        ];
    }
}
```